components/select/nz-select-top-control.component.ts

```diff
@@ -36,7 +36,12 @@ import { NzSelectService } from './nz-select.service';
   animations: [zoomMotion],
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
-  templateUrl: './nz-select-top-control.component.html'
+  templateUrl: './nz-select-top-control.component.html',
+  host: {
+    '[class.ant-select-selection--single]': 'nzSelectService.isSingleMode',
```

Comment:
there is no need to add the class here

Reply:
Yeah this is because I moved these into the component's host https://github.com/NG-ZORRO/ng-zorro-antd/pull/3071/files#diff-290abbcec228b4006c68b790f2878cf5L18
I can change it back in order to be consistent.